tools/deployment/presto-clp/scripts/generate-split-filter-file.py

```diff
@@ -0,0 +1,182 @@
+import argparse
+import json
+import logging
+import sys
+from pathlib import Path
+from typing import Dict, Final, List, Optional, TypedDict
+
+# CONSTANTS
+ANSI_BOLD: Final[str] = "\033[1m"
+ANSI_RESET: Final[str] = "\033[0m"
+DEFAULT_TIMESTAMP_KEY: Final[str] = "timestamp"
+DEFAULT_REQUIRED: Final[bool] = False
+
+# Set up console logging
+logging_console_handler = logging.StreamHandler()
+logging_formatter = logging.Formatter(
+    "%(asctime)s.%(msecs)03d %(levelname)s [%(module)s] %(message)s", datefmt="%Y-%m-%dT%H:%M:%S"
+)
+logging_console_handler.setFormatter(logging_formatter)
+
+# Set up root logger
+root_logger = logging.getLogger()
+root_logger.setLevel(logging.INFO)
+root_logger.addHandler(logging_console_handler)
+
+# Create logger
+logger = logging.getLogger(__name__)
+
+
+class RangeMapping(TypedDict):
+    lowerBound: str
+    upperBound: str
+
+
+class CustomOptions(TypedDict):
+    rangeMapping: RangeMapping
+
+
+class SplitFilterRule(TypedDict):
+    columnName: str
+    customOptions: CustomOptions
+    required: bool
+
+
+SplitFilterDict = Dict[str, List[SplitFilterRule]]
+DEFAULT_RANGE_MAPPING: Final[RangeMapping] = {
+    "lowerBound": "begin_timestamp",
+    "upperBound": "end_timestamp",
+}
+DEFAULT_CUSTOM_OPTIONS: Final[CustomOptions] = {"rangeMapping": DEFAULT_RANGE_MAPPING}
+
+
+def main(argv=None) -> int:
+    if argv is None:
+        argv = sys.argv
+
+    args_parser = argparse.ArgumentParser(
+        description="Generates a split filter file for all user-configured datasets."
+    )
+    args_parser.add_argument(
+        "--clp-package-dir", help="CLP package directory.", required=True, type=Path
+    )
+    args_parser.add_argument(
+        "--output-file", help="Path for the split filter file.", required=True, type=Path
+    )
+
+    parsed_args = args_parser.parse_args(argv[1:])
+    clp_package_dir: Path = parsed_args.clp_package_dir.resolve()
+    archives_dir = clp_package_dir / "var" / "data" / "archives"
+    json_output_file: Path = parsed_args.output_file.resolve()
+    out_dir = json_output_file.parent
+
+    if not archives_dir.exists():
+        logger.error("Archives directory does not exist: %s", archives_dir)
+        return 1
+    if not archives_dir.is_dir():
+        logger.error("Archives path is not a directory: %s", archives_dir)
+        return 1
+    if not out_dir.exists():
+        logger.error("Output directory does not exist: %s", out_dir)
+        return 1
+    if not out_dir.is_dir():
+        logger.error("Output parent is not a directory: %s", out_dir)
+        return 1
```

Comment:
We can have a util function for these at top of the file, so we don't duplicate the same logic twice.